pkg/internal/limitrange/transformer.go

```diff
@@ -115,19 +127,29 @@ func getDefaultAppContainerRequest(limitRange *corev1.LimitRange, nbContainers i
 				if item.Min != nil {
 					min = item.Min[name]
 				}
+
+				var result resource.Quantity
 				if name == corev1.ResourceMemory || name == corev1.ResourceEphemeralStorage {
-					r[name] = takeTheMax(request, *resource.NewQuantity(defaultRequest.Value()/int64(nbContainers), defaultRequest.Format), min)
+					result = takeTheMax(request, *resource.NewQuantity(defaultRequest.Value()/int64(nbContainers), defaultRequest.Format), min)
 				} else {
-					r[name] = takeTheMax(request, *resource.NewMilliQuantity(defaultRequest.MilliValue()/int64(nbContainers), defaultRequest.Format), min)
+					result = takeTheMax(request, *resource.NewMilliQuantity(defaultRequest.MilliValue()/int64(nbContainers), defaultRequest.Format), min)
+				}
+				// only set non-zero request values
+				if !isZero(result) {
+					r[name] = result
 				}
 			}
 		}
 	}
+	// return nil if the resource list is empty to avoid setting an empty defaultrequest
```

Comment:
NIT: should this be done for the other two other `get` functions - `getDefaultContainerRequest` and `getDefaultLimits`? I havent looked at whether its needed or not but just wondering since we added this here.